src/main/java/org/prebid/server/bidder/smartadserver/SmartadserverBidder.java

```diff
@@ -59,13 +60,20 @@ public SmartadserverBidder(String endpointUrl, JacksonMapper mapper) {
         for (Imp imp : request.getImp()) {
             try {
                 final ExtImpSmartadserver extImpSmartadserver = parseImpExt(imp);
+                final String networkId = String.valueOf(extImpSmartadserver.getNetworkId());
+                final Optional<Site> siteOpt = Optional.ofNullable(request.getSite());
+                final Publisher publisher = siteOpt.flatMap(site -> Optional.ofNullable(site.getPublisher()))
+                        .map(p -> p.toBuilder())
+                        .orElse(Publisher.builder())
+                        .id(networkId)
+                        .build();
+                final Site site = siteOpt
+                        .map(s -> s.toBuilder())
+                        .orElse(Site.builder())
+                        .publisher(publisher).build();
```

Comment:
I'm coming from Scala with a focus on functional styles. Is this kind of programming style okay for this project?

Reply:
Usually we prefer to use methods like modify[DesiredObject], like this

```java
private Site modifySite(Site site, Integer networkId) {
        final Site.SiteBuilder siteBuilder = site != null ? site.toBuilder() : Site.builder();
        final Publisher sitePublisher = site != null ? site.getPublisher() : null;

        return siteBuilder.publisher(modifyPublisher(sitePublisher, networkId)).build();
    }

    private Publisher modifyPublisher(Publisher publisher, Integer networkId) {
        final Publisher.PublisherBuilder publisherBuilder = publisher != null
                ? publisher.toBuilder()
                : Publisher.builder();

        return publisherBuilder.id(String.valueOf(networkId)).build();
    }
```
Then your bidRequest can be created like 
```java
final BidRequest updatedRequest = request.toBuilder()
                        .imp(Collections.singletonList(imp))
                        .site(modifySite(request.getSite(), extImpSmartadserver.getNetworkId()))
                        .build();
```

Reply:
I refactored according to your suggestions :smiley: 

Out of curiosity. Was not using `Optional` an explicit decisions and if why so? Performance, too much refactoring? I was thinking about opening an issue to discuss this as I'm stumbling over NPEs quite a lot :joy: (not used to `null` anymore)